go/vt/vtgate/planbuilder/testdata/filter_cases.json

```diff
@@ -1966,7 +1966,7 @@
               "Sharded": true
             },
             "FieldQuery": "select id from `user` where 1 != 1",
-            "Query": "select id from `user` where not :__sq_has_values and id not in ::__sq1",
+            "Query": "select id from `user` where not :__sq_has_values or id not in ::__sq1",
```

Comment:
I have a question. It seems that in this scenario, we can remove the "not :__sq_has_values" condition. Why do we need to convert it to "or"?

Reply:
Excellent question!

We have already built the queries by this time, so we need to be able to send down something that is valid SQL. 
`NOT IN ()` does not even parse - we need to have at least one value there. 
We can't stick `NULL` there, `NOT IN (NULL)` evaluates to `NULL`, not `true`.
We just put in the fake value 0. Just to make sure that we end up with `true` for the empty results, even if the column we are comparing against does contain a `0`, we add this `not :__sq_has_values` that will return true no matter what.

Reply:
If it is important, we would need to come up with a value that we put into the `::__sq1` tuple that guarantees that `id not in ::__sq1` returns true for all values of `id`. We can make it very unlikely, but not guarantee that we will no clash, right?